cmd/crane/cmd/pull.go

```diff
@@ -81,20 +85,42 @@ func NewCmdPull(options *[]crane.Option) *cobra.Command {
 					return fmt.Errorf("saving legacy tarball %s: %w", path, err)
 				}
 			case "oci":
-				if err := crane.MultiSaveOCI(imageMap, path); err != nil {
-					return fmt.Errorf("saving oci image layout %s: %w", path, err)
-				}
-
-				// crane.MultiSaveOCI doesn't support index, so just append these at the end.
+				// Don't use crane.MultiSaveOCI so we can control annotations.
 				p, err := layout.FromPath(path)
 				if err != nil {
-					return err
+					p, err = layout.Write(path, empty.Index)
+					if err != nil {
+						return err
+					}
 				}
+				for ref, img := range imageMap {
+					opts := []layout.Option{}
+					if annotateRef {
+						parsed, err := name.ParseReference(ref, o.Name...)
+						if err != nil {
+							return err
+						}
+						opts = append(opts, layout.WithAnnotations(map[string]string{
+							"org.opencontainers.image.ref.name": parsed.Name(),
```

Comment:
> No semantic restriction is given for the "org.opencontainers.image.ref.name" annotation of descriptors.

but also

> Implementor's Note: A common use case of descriptors with a "org.opencontainers.image.ref.name" annotation is representing a "tag" for a container image. For example, an image may have a tag for different versions or builds of the software. In the wild you often see "tags" like "v1.0.0-vendor.0", "2.0.0-debug", etc. Those tags will often be represented in an image-layout repository with matching "org.opencontainers.image.ref.name" annotations like "v1.0.0-vendor.0", "2.0.0-debug", etc.

Despite the spec asserting otherwise, do tools that interact with layouts tend to interpret these as tags, and not full image refs?

If so this would have been a problem before too, just curious if we're moving even further from accepted behavior.

Reply:
It's straightforward to distinguish a tag from a full ref (`strings.Contains(v, "/")`), so I feel like it _should_ be okay to do this, but it could possibly break some stuff (which is why I don't want to do it by default anymore).

The grammar defined in https://github.com/opencontainers/image-spec/blob/main/annotations.md allows for a full ref, so I don't feel like we'd be doing anything particularly egregious, but @sudo-bmitch might disagree.